File: android/vbt-app/app/src/main/java/com/vbt/app/domain/usecase/CalculatePowerUseCase.kt
package com.vbt.app.domain.usecase

import javax.inject.Inject

class CalculatePowerUseCase @Inject constructor() {
    fun calculateMeanPower(loadKg: Float, distanceM: Float, durationMs: Int): Float {
        if (durationMs <= 0) return 0f
        val force = loadKg * 9.81f
        val meanVelocity = distanceM / (durationMs / 1000f)
        return force * meanVelocity
    }

    fun calculateMeanPower(loadKg: Float, meanVelocityMs: Float): Float {
        val force = loadKg * 9.81f
        return force * meanVelocityMs
    }

    fun calculatePeakPower(loadKg: Float, peakVelocityMs: Float): Float {
        val force = loadKg * 9.81f
        return force * peakVelocityMs
    }
}

File: ios/VBT/VBT/Domain/UseCases/CalculatePowerUseCase.swift
import Foundation

/// Port 1:1 z Android `domain/usecase/CalculatePowerUseCase.kt`.
struct CalculatePowerUseCase {
    func meanPower(loadKg: Float, distanceM: Float, durationMs: Int) -> Float {
        guard durationMs > 0 else { return 0 }
        let force = loadKg * 9.81
        let meanVelocity = distanceM / (Float(durationMs) / 1000)
        return force * meanVelocity
    }

    func meanPower(loadKg: Float, meanVelocityMs: Float) -> Float {
        let force = loadKg * 9.81
        return force * meanVelocityMs
    }

    func peakPower(loadKg: Float, peakVelocityMs: Float) -> Float {
        let force = loadKg * 9.81
        return force * peakVelocityMs
    }
}
